lib/cib/cib_ops.c

```diff
@@ -612,8 +612,8 @@ update_cib_object(xmlNode * parent, xmlNode * update)
               pcmk__s(object_id, ""),
               ((object_id == NULL)? "" : "'"));
 
-    for (a_child = pcmk__xml_first_child(update); a_child != NULL;
-         a_child = pcmk__xml_next(a_child)) {
+    for (a_child = pcmk__xe_first_child(update); a_child != NULL;
+         a_child = pcmk__xe_next(a_child)) {
```

Comment:
I'm trying to trace through these more closely, and I give up on this one (and the next). It looks OK but I'm not positive. What was your reasoning on this one?

Reply:
For the first one, two things:  (1) We call `pcmk__xe_id(a_child)` which led me to believe it's something that would have an attribute.  I haven't checked libxml, but it seems likely that only element nodes would have an ID attribute.  It's true that function could return NULL if there's no ID (which could also mean it doesn't have attributes?), but in that case... (2) When we log a_child, we always log it as `<%s>, a_child->name`.  The combination makes me think we are already treating it like it's an element.

For the second one:  In addition to calling `pcmk__xe_id` on it in `add_cib_object`, we also pass it to `update_cib_object` so if the above paragraph means it's always an element, then the same should apply here.

Reply:
Calling `pcmk__xe_id()` (or getting any other attribute) isn't enough unless we're requiring it to be non-NULL. Using `->name` is pretty definitive (I don't think comment or text nodes have names), but in this case it's only used for logging, so there's still room for doubt. It definitely looks *designed* for elements, but it might accidentally correctly handle non-elements (which has happened before). I'm leaning to reverting this one, without tracing all the way backward to how the XML is created. It seems to go back to the `input` argument of `cib_process_create()`, and that sounds pretty generic.

Reply:
Reverted the portions in this file.